modules/frontend/frontend.go

```diff
@@ -30,19 +26,24 @@ const (
 	apiPathSearch = "/api/search"
 )
 
-// NewTripperware returns a Tripperware configured with a middleware to route, split and dedupe requests.
-func NewTripperware(cfg Config, apiPrefix string, logger log.Logger, registerer prometheus.Registerer) (queryrange.Tripperware, error) {
-	level.Info(logger).Log("msg", "creating tripperware in query frontend")
+// NewMiddleware returns a Middleware configured with a middleware to route, split and dedupe requests.
+func NewMiddleware(cfg Config, apiPrefix string, logger log.Logger, registerer prometheus.Registerer) (Middleware, error) {
+	level.Info(logger).Log("msg", "creating middleware in query frontend")
 
-	tracesTripperware := NewTracesTripperware(cfg, logger, registerer)
-	searchTripperware := NewSearchTripperware()
+	// todo: push this farther down? into the actual creation of the shardingware?
```

Comment:
I think the individual components should be responsible for checking their config on startup and returning an error. It makes more sense to me that the "thing" that uses the config is responsible for rejecting it if it is misconfigured.

Failing fast doesn't really matter here. A few microseconds difference on startup is not meaningful.

Reply:
Ok by fast I meant "before doing a bunch of other work". We start up a lot of other modules, may blocklist polling in the future, and all of this work can be avoided by checking for the right config value early. But I guess it would be a bigger deal for a component like the ingester, its fine here.